### PR TITLE
More Haptic Feedback

### PR DESCRIPTION
**Pull request by Icecovery**
- Added Haptic feedback to all interactable props, ladders, and hatches
- Added haptic tester to KerbalVR_UnitySteamVR project
- Made haptic configs available in the settings file

### CODE REVIEW

File: KerbalVR_Mod/KerbalVR/InteractionCommon/VRFlightStick.cs [hunk 1]
```diff
@@ -152,16 +152,18 @@ private void GetInput(FlightCtrlState st)
 			}
 		}
 
-		private void OnGrab(Hand hand)
+		private void OnGrab(Hand hand, SteamVR_Input_Sources source)
```
Comment (JonnyOThan) [inline KerbalVR_Mod/KerbalVR/InteractionCommon/VRFlightStick.cs:155]: There is Hand.handType for this (maybe needs a better name)

**Review by JonnyOThan — CHANGES_REQUESTED** [review]

File: KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_InteractableBehaviour.cs [hunk 2]
```diff
@@ -30,19 +30,19 @@ public Hand GrabbedHand
...
                     if (lastHand != null && OnRelease != null)
 					{
-						OnRelease.Invoke(lastHand);
+						OnRelease.Invoke(lastHand, lastHand.handType);
 					}
                     lastHand = value;
                     if (value != null && OnGrab != null)
 					{
-                        OnGrab.Invoke(value);
+                        OnGrab.Invoke(value, value.handType);
```
Comment (JonnyOThan) [inline KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_InteractableBehaviour.cs:38]: space/tab indentation is messed up here; I'd prefer to switch to tabs eventually but the original code used spaces.  Please try to keep new stuff consistent with whatever is around it and eventually we can do a single commit that changes it all.

File: KerbalVR_Mod/KerbalVR/InternalModules/KerbalVR_Cover.cs [hunk 3]
```diff
@@ -143,8 +144,10 @@ internal void CoverClose()
         /// <param name="open"><see langword="true"/> to set cover open, <see langword="false"/> otherwise</param>
         public void SetState(bool open)
         {
-            interactionListener.SetAngle(open ? minAngle : maxAngle);
-        }
+			currentAngle = open ? minAngle : maxAngle;
```
Comment (JonnyOThan) [inline KerbalVR_Mod/KerbalVR/InternalModules/KerbalVR_Cover.cs:147]: More indentation mismatch

File: KerbalVR_Mod/KerbalVR/InteractionCommon/VRFlightStick.cs (hunk 1 above)
Comment (Icecovery) [reply KerbalVR_Mod/KerbalVR/InteractionCommon/VRFlightStick.cs]: Fixed in 767ad21.

File: KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_InteractableBehaviour.cs (hunk 2 above)
Comment (Icecovery) [reply KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_InteractableBehaviour.cs]: Fixed in 767ad21.

File: KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_PinchInteractable.cs [hunk 4]
```diff
@@ -13,8 +13,8 @@ public interface IPinchInteractable
 
         GameObject GameObject { get; }
 
-        void OnPinch(Hand hand);
-        void OnHold(Hand hand);
-        void OnRelease(Hand hand);
+        void OnPinch(Hand hand, SteamVR_Input_Sources source);
```
Comment (JonnyOThan) [inline KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_PinchInteractable.cs:16]: InputSource should be removed here

File: KerbalVR_Mod/KerbalVR/InternalModules/KerbalVR_Button.cs [hunk 5]
```diff
@@ -8,118 +8,118 @@
...
-				}
-			}
+                //if (!latchedCover)
+                //{
+                //	HapticUtils.Light(inputSource);
+                //}
+            }
 
-			public void OnExit(Hand hand, Collider buttonCollider, SteamVR_Input_Sources inputSource)
+            public void OnExit(Hand hand, Collider buttonCollider, SteamVR_Input_Sources inputSource)
```
Comment (JonnyOThan) [inline KerbalVR_Mod/KerbalVR/InternalModules/KerbalVR_Button.cs:88]: Could remove inputSource from these as well; looks like I missed that in a previous change.

File: KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_PinchInteractable.cs (hunk 4 above)
Comment (Icecovery) [reply KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_PinchInteractable.cs]: Fixed in 8141003.

File: KerbalVR_Mod/KerbalVR/InternalModules/KerbalVR_Button.cs (hunk 5 above)
Comment (Icecovery) [reply KerbalVR_Mod/KerbalVR/InternalModules/KerbalVR_Button.cs]: Fixed in 8141003.

**Review by JonnyOThan — APPROVED** [review]

**Pull request merged**